tests/reporting/reporting_utils.py

```diff
@@ -0,0 +1,168 @@
+            market.automatedReport([0,2], sender=tester.k0)
+
+    # Fast forward to the reporting phase time
+    reportingWindow = testFixture.applySignature('ReportingWindow', branch.getNextReportingWindow())
+    testFixture.chain.head_state.timestamp = market.getEndTime() + 1
+
+    # This will cause us to be in the AUTOMATED REPORTING phase
+    assert market.getReportingState() == testFixture.constants.AUTOMATED_REPORTING()
+
+def proceedToLimitedReporting(testFixture, market, makeReport, disputer):
+    if (market.getReportingState() != testFixture.constants.AUTOMATED_REPORTING()):
+        proceedToAutomatedReporting(testFixture, market)
+
+    # To proceed to limited reporting we will either dispute an automated report or not make an automated report within the alotted time window for doing so
+    if (makeReport):
+        assert market.automatedReport([0,2], sender=tester.k0)
+        assert market.getReportingState() == testFixture.constants.AUTOMATED_DISPUTE()
+        assert market.disputeAutomatedReport(sender=disputer)
+    else:
+        testFixture.chain.head_state.timestamp = market.getEndTime() + testFixture.constants.AUTOMATED_REPORTING_DURATION_SECONDS() + 1
+
+    # We're in the LIMITED REPORTING phase now
+    assert market.getReportingState() == testFixture.constants.LIMITED_REPORTING()
+
+def proceedToAllReporting(testFixture, market, makeReport, automatedDisputer, limitedDisputer):
+    cash = testFixture.cash
+    branch = testFixture.branch
+    reputationToken = testFixture.applySignature('ReputationToken', branch.getReputationToken())
+    reportingWindow = testFixture.applySignature('ReportingWindow', market.getReportingWindow())
+
+    if (market.getReportingState() != testFixture.constants.LIMITED_REPORTING()):
+        proceedToLimitedReporting(testFixture, market, makeReport, automatedDisputer)
+
+    testFixture.chain.head_state.timestamp = reportingWindow.getDisputeStartTime() + 1
+    assert market.getReportingState() == testFixture.constants.LIMITED_DISPUTE()
+
+    assert market.disputeLimitedReporters(sender=limitedDisputer)
+
+    # We're in the ALL REPORTING phase now
+    assert market.getReportingState() == testFixture.constants.ALL_REPORTING()
+
+def proceedToForking(testFixture, market, makeReport, automatedDisputer, limitedDisputer, reporter):
+    market = testFixture.binaryMarket
+    branch = testFixture.branch
+    reputationToken = testFixture.applySignature('ReputationToken', branch.getReputationToken())
+
+    # Proceed to the ALL REPORTING phase
+    if (market.getReportingState() != testFixture.constants.ALL_REPORTING()):
+        proceedToAllReporting(testFixture, market, makeReport, automatedDisputer, limitedDisputer)
+
+    reportingWindow = testFixture.applySignature('ReportingWindow', market.getReportingWindow())
+
+    reportingTokenNo = testFixture.getReportingToken(market, [2,0])
+    reportingTokenYes = testFixture.getReportingToken(market, [0,2])
```

Comment:
Only the `no` token is used, recommend deleting this line for getting the yes token.